File: frontend/src/services/profileService.ts
import * as ProfileBinding from '../../bindings/changeme/backend/services/profileservice.js'
import type { SettingItem, User } from '../types'
import { mockSettings, mockUser, shouldUseMock } from './mockFallback'

export const profileService = {
    async getCurrentUser(): Promise<User | null> {
        if (shouldUseMock()) return mockUser
        const result = await ProfileBinding.GetCurrentUser()
        return (result ?? null) as unknown as User | null
    },

    async getSettings(): Promise<SettingItem[]> {
        if (shouldUseMock()) return mockSettings
        const result = await ProfileBinding.GetSettings()
        return (result ?? []) as unknown as SettingItem[]
    },
}

File: backend/services/profile_service.go
package services

import (
	"changeme/backend/domain"
	"changeme/backend/repository"
)

// ProfileService 提供个人中心数据
type ProfileService struct {
	repo repository.ProfileRepository
}

func NewProfileService(r repository.ProfileRepository) *ProfileService {
	return &ProfileService{repo: r}
}

// GetCurrentUser 返回当前登录用户信息
func (s *ProfileService) GetCurrentUser() (*domain.User, error) {
	return s.repo.GetCurrentUser()
}

// GetSettings 返回设置项列表
func (s *ProfileService) GetSettings() ([]domain.SettingItem, error) {
	return s.repo.ListSettings()
}
